src/main/java/ch/qos/logback/core/util/ExecutorServiceUtil.java

```diff
@@ -59,21 +59,31 @@ public static synchronized ScheduledExecutorService newScheduledExecutorService(
         return executor;
     }
 
+    /**
+     * Creates threadpool executor suitable for use by logback components.
+     * @return threadpool executor
+     */
+    public static ThreadPoolExecutor newThreadPoolExecutor() {
+        return new ThreadPoolExecutor(CoreConstants.CORE_POOL_SIZE, CoreConstants.MAX_POOL_SIZE, 0L,
+                TimeUnit.MILLISECONDS, new SynchronousQueue<>(), THREAD_FACTORY);
+    }
+
 
     /**
      * Creates an executor service suitable for use by logback components.
-     * @return executor service
+     * @deprecated replaced by {@link #newThreadPoolExecutor()}
      */
     public static ExecutorService newExecutorService() {
-        return new ThreadPoolExecutor(CoreConstants.CORE_POOL_SIZE, CoreConstants.MAX_POOL_SIZE, 0L,
-                TimeUnit.MILLISECONDS, new SynchronousQueue<>(), THREAD_FACTORY);
+        return newThreadPoolExecutor();
     }
 
     /**
      * Shuts down an executor service.
      * @param executorService the executor service to shut down
      */
     public static void shutdown(ExecutorService executorService) {
-        executorService.shutdownNow();
+        if (executorService != null) {
+            executorService.shutdownNow();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Call `shutdown()` to reject any incoming tasks and `awaitTermination()` to wait for existing tasks to complete or until a timeout is reached. To stop the execution and processing of tasks, if necessary, call `shutdownNow()`. Check the return value of `awaitTermination()` and handle when the method times out while waiting for other threads to stop executing. See the following diffs on GitHub that make this code change: [diff1](https://github.com/glowroot/glowroot/commit/81de5a3699c59e0cf98d485c380c1034e6cd28a5#diff-c608d2c40790b0e0e5228e6647f69729caa756bbe9127169139406314b80897eL143), [diff2](https://github.com/apache/directory-kerby/commit/3ed0e7cd3b8a0c3ce29293153e4e72f422f0637f#diff-eb42f072849f520563ec2a686789a836195f18075cc222ef355c6ea710b0a360L84)

Learn more: [ExecutorService in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html)